src/client.js

```diff
@@ -42,15 +42,43 @@ export default class Client {
     }, options);
 
     if (options.apiHostname && isString(options.apiHostname)) {
-      const apiHostnameParsed = url.parse(options.apiHostname);
-      options.apiProtocol = apiHostnameParsed.protocol || 'https:';
-      options.apiHost = apiHostnameParsed.host;
+      let apiHostname = options.apiHostname;
+
+      if (/^https?:\/\//i.test(apiHostname) === false) {
```

Comment:
Ah, good point. I didn't read the regex correctly. In that case, I'm ok with your current proposal. Protocols outside of http(s) are not really something we need to handle.